libraries/libmesosphere/source/kern_k_memory_block_manager.cpp

```
@@ -189,43 +188,39 @@ namespace ams::kern {
+                    remaining_pages = (cur_address + remaining_size - it->GetEndAddress()) / PageSize;
+                    cur_address = it->GetEndAddress();
                 }
             } else {
                 /* If we need to, create a new block before and insert it. */
-                if (cur_info.GetAddress() != GetInteger(cur_address)) {
+                if (it->GetAddress() != GetInteger(cur_address)) {
                     KMemoryBlock *new_block = allocator->Allocate();
 
                     it->Split(new_block, cur_address);
                     it = m_memory_block_tree.insert(*new_block);
                     it++;
```

cppcheck

[performance] Prefer prefix ++/-- operators for non-primitive types.

(postfixOperator)